src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -3417,13 +3417,39 @@ private void fixupCodeBeforeSaving(Command ready)
                name_.getValue().equals("DESCRIPTION");
 
          if (dontStripWhitespace)
+         {
             stripTrailingWhitespace = false;
+         }
 
          if (stripTrailingWhitespace)
          {
+            Pattern pattern = Pattern.create("[ \t]+$", "");
             String code = docDisplay_.getCode();
-            Pattern pattern = Pattern.create("[ \t]+$");
-            String strippedCode = pattern.replaceAll(code, "");
+            String strippedCode = "";
+            
+            // If this is being performed as part of an autosave, avoid
+            // mutating lines containing the cursor.
+            if (isAutoSaving())
+            {
+               int startRow = docDisplay_.getSelectionStart().getRow();
+               int endRow = docDisplay_.getSelectionEnd().getRow();
```

Comment:
Does this detect the multi-line cursor mode (Ctrl+Alt+Up/Down Arrow) along with multi-line selection (I have no idea if that's the same thing to `getSelectionStart/End`)?

Reply:
Good question -- it looks like this only applies to the "primary" cursor / selection. Should I do the legwork to support multiple cursors here?

Reply:
> Should I do the legwork to support multiple cursors here?

Probably not a very high priority, was just curious.